service/worker/scanner/build_ids/scavenger.go

```diff
@@ -267,13 +269,32 @@ func (a *Activities) processUserDataEntry(
 func (a *Activities) findBuildIdsToRemove(
 	ctx context.Context,
 	rateLimiter quotas.RateLimiter,
+	input BuildIdScavangerInput,
 	heartbeat heartbeatDetails,
 	ns *namespace.Namespace,
 	entry *persistence.TaskQueueUserDataEntry,
 ) ([]string, error) {
+	// Only consider build ids that have been active at least as long as the retention time.
+	// This assumes that when a build id is added, it's used soon afterwards.
+	// This lets us avoid making visibility queries that would probably find some workflows.
+	retention := ns.Retention()
+	// Don't consider build ids that were recently the default, since there may be workers
+	// still processing tasks or data that hasn't made it to visibility yet.
+	removableBuildIdDurationSinceDefault := a.removableBuildIdDurationSinceDefault()
+
+	now := func() time.Time {
```

Comment:
You don't need this, you can control the user data entries in the test, create them with a clock that works for the test.